auslib/db.py

```diff
@@ -1487,11 +1487,22 @@ def getPotentialRequiredSignoffs(self, affected_rows, transaction=None):
     def _matchesRegex(self, foo, bar):
         # Expand wildcards and use ^/$ to make sure we don't succeed on partial
         # matches. Eg, 3.6* matches 3.6, 3.6.1, 3.6b3, etc.
-        test = foo.replace('.', '\.').replace('*', '.*')
-        test = '^%s$' % test
-        if re.match(test, bar):
-            return True
-        return False
+        if foo.endswith('*'):
+            if(re.match('^.{0,2}\*$', foo)):
```

Comment:
It's a great idea to limit the globbing to strings of a certain length, but I would suggest a couple of changes:
* Use a simple len() comparison to determine if the string is long enough. Let's limit globbing to strings that have at least two other characters, so this can be "if len(foo) >= 3" (2 characters + a *, which we already know is the final character).
* If this check fails, we should go into else block instead of returning False.